src/encoder.rs

```diff
@@ -1209,11 +1209,18 @@ pub fn encode_tx_block<T: Pixel, W: Writer>(
       residual,
       &ts.input_tile.planes[p].subregion(area),
       &rec.subregion(area),
-      visible_tx_w,
+      tx_size.width(),
       visible_tx_h,
     );
+    if visible_tx_w < tx_size.width() {
+      for row in residual.chunks_mut(tx_size.width()).take(visible_tx_h) {
+        for v in &mut row[visible_tx_w..] {
+          *v = 0;
+        }
+      }
+    }
   }
-  let visible_area = visible_tx_w * visible_tx_h;
+  let visible_area = tx_size.width() * visible_tx_h;
   for a in residual[visible_area..].iter_mut() {
```

Comment:
This for loop initializes the whole tx_size residual[ ] as zeros, correct? If so, isn't the computed diff (i.e. residue) values erased? So, I think this for loop can be put before diff( ) is performed, i.e. between line 1207 and 1208, and
lines 1215~1221 can be removed (I think this block of code is zeroing right side of non-visible area in tx area?)

Reply:
This loop zeroes the area below the visible region, to the full width of the transform.